src/hotspot/share/services/heapDumper.cpp

```diff
@@ -2052,10 +2129,9 @@ void DumpMerger::do_merge() {
 
   // Merge the content of the remaining files into base file. Regardless of whether
   // the merge process is successful or not, these segmented files will be deleted.
-  char path[JVM_MAXPATHLEN];
   for (int i = 0; i < _dump_seq; i++) {
-    memset(path, 0, JVM_MAXPATHLEN);
-    os::snprintf(path, JVM_MAXPATHLEN, "%s.p%d", _path, i);
+    ResourceMark rm;
+    const char* path = get_writer_path(_path, i);
```

Comment:
It's not clear to me why you've chosen to move from a JVM_MAXPATHLEN stack buffer to a dynamically allocated one.

Reply:
It's better to have a single place to generate path of the segment file, so the fix introduced  DumpMerger::get_writer_path()
And theoretically the path can be longer than JVM_MAXPATHLEN